### PR TITLE
DBZ-1773 Process TINYINT(1) as number

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-1773

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Test failure looks related.

**Comment by jpechane** [comment]
@gunnarmorling Fixed, looks like another JDBC driver quirk

**Comment by gunnarmorling** [comment]
@jpechane, how does this impact `BOOLEAN` columns? Will there values be exported as 0/1 now instead of false/true?

**Comment by jpechane** [comment]
@gunnarmorling There are alread exported as 0/1, see https://issues.redhat.com/browse/DBZ-1074

**Comment by gunnarmorling** [comment]
Ah, yes. Remembering now. It seems then the docs need an update, as [they claim](https://debezium.io/documentation/reference/1.1/assemblies/cdc-mysql-connector/as_overview-of-how-the-mysql-connector-works.html#how-the-mysql-connector-maps-data-types_debezium), as they claim that boolean columns would be exported as `BOOLEAN` field.

**Comment by jpechane** [comment]
@gunnarmorling I don't thinks that's that easy, it is sometimes BOOLEAN - when read from binlog and sometimes INT16 - when read from snapshot so I'd rather handle it in a separate PR

**Comment by jpechane** [comment]
@gunnarmorling https://issues.redhat.com/browse/DBZ-1801

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java [hunk 1]
```diff
@@ -160,6 +160,14 @@ else if (actualColumn.jdbcType() == Types.DATE) {
         else if (actualColumn.jdbcType() == Types.TIMESTAMP) {
             return readTimestampField(rs, fieldNo, actualColumn, actualTable);
         }
+        // JDBC's rs.GetObject() will return a Boolean for all TINYINT(1) columns.
+        // TINYINT columns are reprtoed as SMALLINT by JDBC driver
+        else if (actualColumn.jdbcType() == Types.TINYINT || actualColumn.jdbcType() == Types.SMALLINT) {
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java]: Ok, read up a bit more on this. Seems we could achieve the same without any real code change [by setting](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html) `tinyInt1isBit` to false (it's true by default). Both should work, leaving it to you what you prefer. In any case it should be a workaround for the issue reporter until the fix has been shipped.

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlTinyIntIT.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    private void assertChangeRecord() throws InterruptedException {
+        final SourceRecord record = consumeRecord();
+        Assertions.assertThat(record).isNotNull();
+        final Struct change = ((Struct) record.value()).getStruct("after");
+
+        Assertions.assertThat(change.getInt16("ti")).isEqualTo((short) 100);
+        Assertions.assertThat(change.getInt16("ti1")).isEqualTo((short) 5);
+        Assertions.assertThat(change.getInt16("ti2")).isEqualTo((byte) 50);
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlTinyIntIT.java]: How comes the expected type differs between short and byte? As per the docs, it should always be `int8`? The `N` in `TINYINT(N)` is only the display width, it doesn't impact the value range AFAIU.

Comment (gunnarmorling) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlTinyIntIT.java]: @jpechane, any thoughts on this?

Comment (jpechane) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlTinyIntIT.java]: @gunnarmorling Sorry, was a mistake on my side, fixed. In fact it does not matter if value passed is short or byte as it is alway casted to short according to method parameter. But good to align for consistency

Comment (gunnarmorling) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlTinyIntIT.java]: Ok, thanks!

**Comment by gunnarmorling** [comment]
Applied, thanks!

**Pull request merged**